core/templates/services/playthrough.service.ts

```diff
@@ -17,383 +17,268 @@
+  private playthrough?: Playthrough = null;
+  private stopwatch?: Stopwatch = null;
+  private expDurationInSecs?: number = null;
+
+  constructor(
+      private explorationFeaturesService: ExplorationFeaturesService,
+      private learnerActionObjectFactory: LearnerActionObjectFactory,
+      private playthroughBackendApiService: PlaythroughBackendApiService,
+      private playthroughObjectFactory: PlaythroughObjectFactory,
+      private stopwatchObjectFactory: StopwatchObjectFactory) {}
+
+  /**
+   * The ordering of checks in this method prioritizes the following types of
+   * playthroughs:
+   *    1. MultipleIncorrectSubmissionsIssue
+   *    2. CyclicStateTransitionsIssue
+   *    3. EarlyQuitIssue
+   */
+  private classifyPlaythrough(): void {
+    if (this.misTracker !== null && this.misTracker.isIssue()) {
+      this.playthrough.issueType = (
+        AppConstants.ISSUE_TYPE_MULTIPLE_INCORRECT_SUBMISSIONS);
+      this.playthrough.issueCustomizationArgs = (
+        <IMultipleIncorrectSubmissionsCustomizationArgs>{
+          state_name: {value: this.misTracker.currentStateName},
+          num_times_answered_incorrectly: {value: this.misTracker.numTries}
+        });
+    } else if (this.cstTracker !== null && this.cstTracker.isIssue()) {
+      this.playthrough.issueType = (
+        AppConstants.ISSUE_TYPE_CYCLIC_STATE_TRANSITIONS);
+      this.playthrough.issueCustomizationArgs = (
+        <ICyclicStateTransitionsCustomizationArgs>{
+          state_names: {value: this.cstTracker.cycle}
+        });
+    } else if (this.eqTracker !== null && this.eqTracker.isIssue()) {
+      this.playthrough.issueType = AppConstants.ISSUE_TYPE_EARLY_QUIT;
+      this.playthrough.issueCustomizationArgs = (
+        <IEarlyQuitCustomizationArgs>{
+          state_name: {value: this.eqTracker.stateName},
+          time_spent_in_exp_in_msecs: {value: this.expDurationInSecs * 1000}
+        });
     }
+  }
 
-    recordExplorationStartAction(initStateName: string): void {
-      if (this.isPlaythroughDiscarded()) {
-        return;
-      }
-      let expStartLearnerAction = this.learnerActionObjectFactory.createNew(
-        AppConstants.ACTION_TYPE_EXPLORATION_START,
-        {
-          state_name: {
-            value: initStateName
-          }
-        },
-        ServicesConstants.CURRENT_ACTION_SCHEMA_VERSION);
+  private isPlaythroughRecordingEnabled(): boolean {
+    return (
+      this.explorationFeaturesService.isPlaythroughRecordingEnabled() &&
+      this.isLearnerInSamplePopulation === true);
+  }
 
-      PlaythroughService.playthrough.actions.unshift(expStartLearnerAction);
+  private isLearnerJustBrowsing(): boolean {
+    return (
+      // Learners who never enter an answer are probably just browsing.
+      !this.playthrough.actions.some(a => a.actionType === 'AnswerSubmit') ||
+      // Learners who leave the exploration quickly are probably just browsing.
+      this.expDurationInSecs < 45);
```

Comment:
Btw, I just realized -- we call it "early quit", but it seems we are looking at arbitrarily late quits?

Reply:
Eh, not really. I don't think the time since the last state transition is particularly relevant to anything.

Maybe < 5 mins?

Reply:
Sure, that works too. I'll go ahead and make the changes.

Reply:
Sure, that works too.

Just so we're on the same page, the proposal is:
```typescript
isLearnerJustBrowsing() => timeSpentInExpSecs < 45
foundEarlyQuitIssue => timeSpentInExpSecs < 300
```
right?

Reply:
Yes this seems fine. Please make 45 a named configurable constant though.